lib/cocoapods/command/spec/cat.rb

```diff
@@ -39,6 +41,11 @@ def run
                        specs = get_path_of_spec(query, @show_all).split(/\n/)
                        index = UI.choose_from_array(specs, "Which spec would you like to print [1-#{specs.count}]? ")
                        specs[index]
+                     elsif @version
+                       specs = get_path_of_spec(query, true).split(/\n/)
+                       matching_spec = specs.select { |p| %r{(#{@version}\/)} =~ p }.first
```

Comment:
This will also match version 12.3.4 if the requested version is 2.3.4…

Imho it would be better to adapt the get_path_for_spec method so that its 2nd parameter can accept a string as well as a boolean, and if it is a string, only return the path for that version.

Would feel safer than relying on a regex on the resulting path, especially since that makes assumptions on how the path is formed, and if the convention on this changes in a future version of CocoaPods (like it did when we sharded the spec repo a couple of years ago) that would break.